testproject/Assets/Tests/Runtime/MultiprocessRuntime/BaseMultiprocessTests.cs

```diff
@@ -149,20 +171,32 @@ public virtual IEnumerator Setup()
             // Moved this out of OnSceneLoaded as OnSceneLoaded is a callback from the SceneManager and just wanted to avoid creating
             // processes from within the same callstack/context as the SceneManager.  This will instantiate up to the WorkerCount and
             // then any subsequent calls to Setup if there are already workers it will skip this step
-            if (NetworkManager.Singleton.ConnectedClients.Count - 1 < WorkerCount)
+            if (!m_LaunchRemotely)
             {
-                var numProcessesToCreate = WorkerCount - (NetworkManager.Singleton.ConnectedClients.Count - 1);
-                for (int i = 1; i <= numProcessesToCreate; i++)
+                if (NetworkManager.Singleton.ConnectedClients.Count - 1 < WorkerCount)
                 {
-                    MultiprocessLogger.Log($"Spawning testplayer {i} since connected client count is {NetworkManager.Singleton.ConnectedClients.Count} is less than {WorkerCount} and Number of spawned external players is {MultiprocessOrchestration.ActiveWorkerCount()} ");
-                    string logPath = MultiprocessOrchestration.StartWorkerNode(); // will automatically start built player as clients
-                    MultiprocessLogger.Log($"logPath to new process is {logPath}");
-                    MultiprocessLogger.Log($"Active Worker Count {MultiprocessOrchestration.ActiveWorkerCount()} and connected client count is {NetworkManager.Singleton.ConnectedClients.Count}");
+                    var numProcessesToCreate = WorkerCount - (NetworkManager.Singleton.ConnectedClients.Count - 1);
+                    for (int i = 1; i <= numProcessesToCreate; i++)
+                    {
+                        MultiprocessLogger.Log($"Spawning testplayer {i} since connected client count is {NetworkManager.Singleton.ConnectedClients.Count} is less than {WorkerCount} and Number of spawned external players is {MultiprocessOrchestration.ActiveWorkerCount()} ");
+                        string logPath = MultiprocessOrchestration.StartWorkerNode(); // will automatically start built player as clients
+                        MultiprocessLogger.Log($"logPath to new process is {logPath}");
+                        MultiprocessLogger.Log($"Active Worker Count {MultiprocessOrchestration.ActiveWorkerCount()} and connected client count is {NetworkManager.Singleton.ConnectedClients.Count}");
+                    }
                 }
             }
-            else
+            else if (m_LaunchRemotely)
             {
-                MultiprocessLogger.Log($"No need to spawn a new test player as there are already existing processes {MultiprocessOrchestration.ActiveWorkerCount()} and connected clients {NetworkManager.Singleton.ConnectedClients.Count}");
+                var launchProcessList = new List<Process>();
+                if (NetworkManager.Singleton.ConnectedClients.Count - 1 < GetWorkerCount())
```

Comment:
what happens if that's not the case, we fail silently?

Reply:
In the near future there will be an option where we can reuse connected clients and not stop and start them each time. The quick start/stop routine has proven to be super unstable since the host doesn't respond quickly enough before the next start up cycle and then things get very muddy.